octoprint_prettygcode/static/js/prettygcode.js

```diff
@@ -132,11 +331,49 @@ $(function () {
                     }else{
                         parserCurState.extrude=false;
                     }
-
                     parserCurState.layerLineNumber =parserLayerLineNumber;
-                    
-                    buffer.push(parserCurState.clone());
+
+                    // if this is a g0/g1, push the state to the buffer
+                    if (is_g0_g1) {buffer.push(parserCurState.clone());}
+                    else{
+                        // This is a g2/g3, so we need to do things a bit differently.
+                        // TODO: Do I need to convert axis mode here, or will things just work out?
+
+                        // is_clockwise
+
+                        // Extract I and J, R, and is_clockwise
+                        var is_clockwise = cmd.indexOf(" G2")>-1;
+                        var i = parseFloat(cmd.split("I")[1]);
+                        var j = parseFloat(cmd.split("J")[1]);
+                        var r = parseFloat(cmd.split("J")[1]);
```

Comment:
Shouldn't this be cmd.split("R")?

Reply:
Yes it should.  Didn't catch that since R isn't implemented.  Good catch